cli/cmd/plugin/unmanaged-cluster/cmd/configure.go

```diff
@@ -37,6 +37,9 @@ func init() {
 	ConfigureCmd.Flags().StringVar(&co.servicecidr, "service-cidr", "", "The CIDR to use for Service IP addresses. Default and format is '10.96.0.0/16'")
 	ConfigureCmd.Flags().Bool("tty-disable", false, "Disable log stylization and emojis")
 	ConfigureCmd.Flags().StringSliceVar(&co.additionalRepo, "additional-repo", []string{}, "Addresses for additional package repositories to install")
+	ConfigureCmd.Flags().StringSliceVar(&co.profile, "profile", []string{}, "(experimental) A profile to install. May be specified multiple times. Should be the fully qualified package name or a prefix to a package name found in an installed package repository. Profile mappings supported - profile-name:profile-version:profile-config-file")
+	ConfigureCmd.Flags().StringSliceVar(&co.profileConfigPath, "profile-config-file", []string{}, "(experimental) Optional: path to a profile values yaml file. Uses default values (when available) if not provided. May be specified multiple times. Strings given via this flag are ordered in a queue and are enqueud in the order they are specified and dequeud when missing profile configs are encountered.")
```

Comment:
@jpmcb As I read this, your implementation supports modes of specifying a profile.

```
--profile fluent-bit:1.7.5:path-to-my-config.yaml
```

and

```
--profile fluent-bit --profile-version 1.7.5 --profile-config-file path-to-my-config.yaml
```

If i'm reading the parsing code correctly, these 2 create equivalent outcomes.

I wonder if we're better offering the `--profile` option exclusively.

My initial reasoning is:

1. Unless it's an extremely high-value UX benefit, it might not make sense to have 2 ways to do the same thing.
2. Code complexity is increased a non-trivial amount by satisfying these 2 models.

What are your thoughts? And/or am I overlooking anything?

Reply:
Hmmm good question. Yes, essentially, this UX supports both profile mapping and just the flag. What's unique about this is it also supports mixing those.

So a user could do something like:
```
--profile fluent-bit:1.7.5
--profile-config-file my-config.yaml
```
I think this is a powerful UX that would allow people to drop in and replace single fields in profile mappings like in the above example: the config file is explicitly declared instead of the profile mapping

But you're right: there is increased code complexity and possibly increased effect of user error/unintended behavior. I'm a bit torn because I really like the flexability of having both and being able to mix and match.

Reply:
Chatted, and we definitely want to go with the simpler UX of a single `--profile` flag with profile mappings:

```
--profile fluent-bit:1.7.5:path-to-values.yaml
```

If users don't want to include a version but they do want to include a values yaml file, they can do either:

```
--profile fluent-bit::path-to-values.yaml
```

or we reserve the `latest` keyword to resolve to the latest package found (the default behavior if no version specified)

```
--profile fluent-bit:latest:path-to-values.yaml
```